song-server/src/main/java/bio/overture/song/server/service/ValidationService.java

```diff
@@ -85,6 +90,27 @@ public Optional<String> validate(@NonNull JsonNode payload) {
               "Found Analysis type: name=%s  version=%s",
               analysisType.getName(), analysisType.getVersion()));
 
+      List<String> fileTypes = new ArrayList<>();
+
+      if (analysisType.getOptions() != null && analysisType.getOptions().getFileTypes() != null) {
+        fileTypes = analysisType.getOptions().getFileTypes();
+      }
+
+      if (CollectionUtils.isNotEmpty(fileTypes)) {
+        JsonNode files = payload.get("files");
+        if (files.isArray()) {
+          for (JsonNode file : files) {
+            String fileType = file.get("fileType").asText();
+            String fileName = file.get("fileName").asText();
+            if (!fileTypes.contains(fileType)) {
+              throw new ValidationException(
+                      String.format("%s name is not supported, supported formats are %s"
+                              , fileName
+                              , String.join(", ", fileTypes)));
+            }
+          }
+        }
+      }
```

Comment:
We should pull this logic into its own function to `validateFileType(payload, analysisType)`, or at very least leave some comments to explain what this section of code is doing.

Also, lets do this after we have run `validateWithSchema()` since that will ensure the analysis has the correct properties at all.